mm2src/mm2_main/tests/docker_tests/tendermint_tests.rs

```diff
@@ -465,222 +521,167 @@ fn test_tendermint_tx_history() {
     use std::convert::TryFrom;
+    use std::str::FromStr;
+    use std::sync::Mutex;
     use std::{env, thread};
 
     const BOB_PASSPHRASE: &str = "iris test seed";
     const ALICE_PASSPHRASE: &str = "iris test2 seed";
 
-    // https://academy.binance.com/en/articles/connecting-metamask-to-binance-smart-chain
-    const TBNB_URLS: &[&str] = &["https://data-seed-prebsc-1-s1.binance.org:8545/"];
-    // https://testnet.bscscan.com/address/0xb1ad803ea4f57401639c123000c75f5b66e4d123
-    const TBNB_SWAP_CONTRACT: &str = "0xB1Ad803ea4F57401639c123000C75F5B66E4D123";
-
-    #[test]
-    fn swap_usdc_ibc_with_nimda() {
-        let bob_passphrase = String::from(BOB_PASSPHRASE);
-        let alice_passphrase = String::from(ALICE_PASSPHRASE);
-
-        let coins = json!([
-            usdc_ibc_iris_testnet_conf(),
-            iris_nimda_testnet_conf(),
-            iris_testnet_conf(),
-        ]);
-
-        let mm_bob = MarketMakerIt::start(
-            json!({
-                "gui": "nogui",
-                "netid": 8999,
-                "dht": "on",
-                "myipaddr": env::var("BOB_TRADE_IP") .ok(),
-                "rpcip": env::var("BOB_TRADE_IP") .ok(),
-                "canbind": env::var("BOB_TRADE_PORT") .ok().map (|s| s.parse::<i64>().unwrap()),
-                "passphrase": bob_passphrase,
-                "coins": coins,
-                "rpc_password": "password",
-                "i_am_seed": true,
-            }),
-            "password".into(),
-            None,
-        )
-        .unwrap();
-
-        thread::sleep(Duration::from_secs(1));
-
-        let mm_alice = MarketMakerIt::start(
-            json!({
-                "gui": "nogui",
-                "netid": 8999,
-                "dht": "on",
-                "myipaddr": env::var("ALICE_TRADE_IP") .ok(),
-                "rpcip": env::var("ALICE_TRADE_IP") .ok(),
-                "passphrase": alice_passphrase,
-                "coins": coins,
-                "seednodes": [mm_bob.my_seed_addr()],
-                "rpc_password": "password",
-                "skip_startup_checks": true,
-            }),
-            "password".into(),
-            None,
-        )
-        .unwrap();
-
-        thread::sleep(Duration::from_secs(1));
-
-        dbg!(block_on(enable_tendermint(
-            &mm_bob,
-            "IRIS-TEST",
-            &["IRIS-NIMDA", "USDC-IBC-IRIS"],
-            IRIS_TESTNET_RPC_URLS,
-            false
-        )));
-
-        dbg!(block_on(enable_tendermint(
-            &mm_alice,
-            "IRIS-TEST",
-            &["IRIS-NIMDA", "USDC-IBC-IRIS"],
-            IRIS_TESTNET_RPC_URLS,
-            false
-        )));
-
-        block_on(trade_base_rel_tendermint(
-            mm_bob,
-            mm_alice,
-            "USDC-IBC-IRIS",
-            "IRIS-NIMDA",
-            1,
-            2,
-            0.008,
-        ));
+    lazy_static! {
+        // Simple lock used for running the swap tests sequentially.
```

Comment:
but why tho?

Reply:
They use same account seed; the same tendermint account tries to swap from multiple instances. Normally, we have this https://github.com/KomodoPlatform/komodo-defi-framework/blob/b0fd99e8406e67ea06435dd028991caa5f522b5c/mm2src/coins/tendermint/tendermint_coin.rs#L891-L919 function which ensures we end up with the correct sequence id so the swap doesn't fail because of sequence mismatch errors, but somehow that didn't help (the node was keep giving the same sequence id instead of the new one, this is either docker testnet specific problem or it's cached for too long from the nucleus node).